src/components/templates/HackathonProjects.js

```diff
@@ -19,19 +19,18 @@ import Button from "../atoms/Button";
 import Rating from "react-rating";
 import emptyStar from "../../assets/star-hollow.png";
 import fullStar from "../../assets/star-full.png";
-import { fetchAllSubmissions } from "../../store/projectSubmission/actions";
+import { useSubmissions } from "../../hooks";
 
 const HackathonProjects = () => {
   const { id } = useParams();
   const { event_title } = useSelector(state =>
     state.events.data.find(event => event.id === Number(id))
   );
-  const { submissions } = useSelector(state => state);
-  const dispatch = useDispatch();
+  const [submissions, fetchSubmissions] = useSubmissions();
```

Comment:
This hook only works when given an event Id
`const [submissions, fetchSubmissions] = useSubmissions(id);`